python/dgl/ops/sddmm.py

```diff
@@ -74,22 +74,10 @@ def gsddmm(g, op, lhs_data, rhs_data, lhs_target='u', rhs_target='v'):
         return gsddmm_internal(
             g._graph, op, lhs_data, rhs_data, lhs_target, rhs_target)
     else:
-        lhs_data_dict = lhs_data
-        rhs_data_dict = rhs_data
-        lhs_list = [None] * g._graph.number_of_ntypes()
-        rhs_list = [None] * g._graph.number_of_ntypes()
-        for srctype, _, dsttype in g.canonical_etypes:
-            src_id = g.get_ntype_id(srctype)
-            dst_id = g.get_ntype_id(dsttype)
-            lhs_data = lhs_data_dict[srctype]
-            rhs_data = rhs_data_dict[dsttype]
-            if op not in ['copy_lhs', 'copy_rhs']:
-                lhs_data, rhs_data = reshape_lhs_rhs(lhs_data, rhs_data)
-            lhs_list[src_id] = lhs_data
-            rhs_list[dst_id] = rhs_data
-        lhs_and_rhs_tuple = tuple(lhs_list + rhs_list)
-        # With max and min reducers infinity will be returned for zero degree nodes
-        return gsddmm_internal_hetero(g, op, lhs_target, rhs_target, *lhs_and_rhs_tuple)
+        # TODO (Israt): Call reshape func
```

Comment:
What does this TODO mean?

Reply:
When number of etype == 1, the reshape function is called to match the dimension of the lhs and rhs data to avoid broadcasting issue. If in practice, nodes in the same heterogeneous graph usually have the same dimensions, its not an issue. But if different nodes have different dimensions, than different etypes may need different broadcasting dims for the same node. Now sure how to handle that case.

Reply:
I see.  Could you put the elaboration as comments just in case we (any of us) forget about what we should do?

Reply:
Done.